File: languages/java/testData/inspection.java
// carets define highlighted elements
package org.jetbrains.iren;

import java.io.IOError;

public class Foo {
    public void boo(int <caret>boi) {
        var <caret>sch = getSomeSchoolNumber();
        var state = Foo.computeSmthg();
        var component = returnMy();
        var answerForTheUltimateQuestionOfLife = computeAnswerForTheUltimateQuestionOfLife();
        var school = getSomeSchoolNumber();
        try {
            System.out.println("Oh My!");
        } catch (RuntimeException ignore) {
        } catch (IOError ignored) {
        } catch (Exception <caret>x) {
        }
    }

    private static State computeSmthg() {
        return new State();
    }

    private int computeAnswerForTheUltimateQuestionOfLife() {
        return 42;
    }

    public int getSomeSchoolNumber(int number<caret>) {
        return 239;
    }

    public MyComponent returnMy() {
        return new MyComponent();
    }

    static class State {
    }

    static class MyComponent {
    }
}

class Foo2 extends Foo {
    @java.lang.Override
    public void boo(int boi) {
        System.out.println("Hello World!");
    }

    @java.lang.Override
    public int getSomeSchoolNumber(int <caret>smth) {
        return 777;
    }
}

File: languages/kotlin/testData/inspection.kt
// carets define highlighted elements
package org.jetbrains.iren;

open class Foo {
    open var x<caret>y = 0
    open fun boo(b<caret>oi: Int) {
        val state = computeSmth()
        val component = returnMy()
        val answerForTheUltimateQuestionOfLife = computeAnswerForTheUltimateQuestionOfLife()
        val school = getSomeSchoolNumber()
        val sc<caret>h = getSomeSchoolNumber()
        try {
            println("Oh My!")
        } catch (ignore: Exception) {
        } catch (ignored: Exception) {
        } catch (_: Exception) {
        } catch (<caret>e: Exception) {
            println("Exception is not ignored")
        }
    }

    private fun computeSmth(): State {
        return State()
    }

    private fun returnMy(): MyComponent {
        return MyComponent()
    }

    private fun computeAnswerForTheUltimateQuestionOfLife(): Int {
        return 42
    }

    private fun getSomeSchoolNumber(): Int {
        return 239
    }

    class State
    class MyComponent
}

internal class Foo2 : Foo() {
    override var xy = 1
    override fun boo(boi: Int) = println("Hello World!")
}
